src/source_clang.cc

```diff
@@ -874,8 +874,8 @@ void Source::ClangViewAutocomplete::async_delete() {
   parse_thread_stop=true;
   delete_thread=std::thread([this](){
     //TODO: Is it possible to stop the clang-process in progress?
-    if(restart_parse_thread.joinable())
-      restart_parse_thread.join();
+    if(full_reparse_thread.joinable())
```

Comment:
Terminate a thread is impossible to perform without leaving objects that have not been freed, as I understand it. Thus this is never part of a thread-library. Processes, on the other hand, can be killed, and their memory is freed by the OS.